File: src/cz/cuni/lf1/lge/ThunderSTORM/filters/ui/EmptyFilterUI.java
package cz.cuni.lf1.lge.ThunderSTORM.filters.ui;

import cz.cuni.lf1.lge.ThunderSTORM.filters.EmptyFilter;
import cz.cuni.lf1.lge.ThunderSTORM.filters.IFilter;
import javax.swing.JPanel;

/**
 *
 * @author Josef Borkovec <josef.borkovec[at]lf1.cuni.cz>
 */
public class EmptyFilterUI implements IFilterUI {

  @Override
  public String getName() {
    return "No filter";
  }

  @Override
  public JPanel getOptionsPanel() {
    return null;
  }

  @Override
  public void readParameters() {
    // nothing to do here
  }

  @Override
  public IFilter getImplementation() {
    return new EmptyFilter();
  }
}

File: src/main/java/cz/cuni/lf1/lge/ThunderSTORM/filters/ui/EmptyFilterUI.kt
package cz.cuni.lf1.lge.ThunderSTORM.filters.ui

import cz.cuni.lf1.thunderstorm.algorithms.filters.Filter
import cz.cuni.lf1.thunderstorm.datastructures.GrayScaleImage

import javax.swing.JPanel

private const val name = "No filter"

public class EmptyFilterUI : FilterUI() {

    public override fun getName()
            = name

    public override fun getOptionsPanel(): JPanel
            = JPanel()

    public override fun getImplementation()
            = object : Filter {
                override fun filter(image: GrayScaleImage) = image
            }

    public override fun createThresholderSymbolTable(): Map<String, (GrayScaleImage) -> GrayScaleImage>
            = mapOf(Pair("Empty.F", { img -> getImplementation().filter(img) }))
}